zooniverse/front-end-monorepo
# app-root, lib-user: Use url search param for stats date range and project state

## Package
- app-root
- lib-user

## Describe your changes
- app-root
  - remove stats date range context
  - remove stats project context
  - add search param for stats start date and end date state to UserStats, Certificate, and GroupStats
  - add search param for stats project state to UserStats, Certificate, and GroupStats
- lib-user
  - refactor UserStats, Certificate, and GroupStats with new dateRange object (includes start date and end date)
  - refactor MainContent for new dateRange object and new Select onChange functions
  - refactor `getDateInterval` utility
  - add `getDateRangeSelectOptions` utility
  - set 'All Time' start date as stats source (user or group) `created_at` date
  - refactor BarChart for new dateRange object
  - refactor `getDateRangeLabel` helper
  - update stories and tests as needed
  - update dev app for stats state per url search param

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
- What user actions should my reviewer step through to review this PR?
  - go to your user stats page
    - change the project
    - change the date range
    - generate a certificate
  - go to a group stats page
    - change the project
    - change the date range
    - go to the full contributors page (no related changes, but should work as expected with stats from all time and all projects)
- Which storybook stories should be reviewed? MainContent, BarChart
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see linked GitHub project

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
